### ISSUE TITLE
init_with=="last" not working

### ISSUE THREAD

**Issue by ChristophNeuner** [killthekitten/kaggle-ds-bowl-2018-baseline#9]
Hello,

I hope you can help me.
First I have trained the head layers only.
Now I want to use the last saved .h5 file and train all layers with a samller learning rate.
But when I change  initialize init_with=="last" in the train.py file the training does not work.

Is there something else I have to change?

Thanks in advance!

Christoph

```
import os

import sys
import random
import math
import time

from bowl_config import bowl_config
from bowl_dataset import BowlDataset
import utils
import model as modellib
from model import log
from glob import glob

import tensorflow as tf
tf.logging.set_verbosity(tf.logging.ERROR)


# Root directory of the project
ROOT_DIR = os.getcwd()

# Directory to save logs and trained model
MODEL_DIR = os.path.join(ROOT_DIR, "logs")

# Local path to trained weights file
COCO_MODEL_PATH = os.path.join(ROOT_DIR, "mask_rcnn_coco.h5")
# Download COCO trained weights from Releases if needed
if not os.path.exists(COCO_MODEL_PATH):
    utils.download_trained_weights(COCO_MODEL_PATH)
    
model = modellib.MaskRCNN(mode="training", config=bowl_config,
                          model_dir=MODEL_DIR)

# Which weights to start with?
init_with = "last"  # imagenet, coco, or last

if init_with == "imagenet":
    model.load_weights(model.get_imagenet_weights(), by_name=True)
elif init_with == "coco":
    # Load weights trained on MS COCO, but skip layers that
    # are different due to the different number of classes
    # See README for instructions to download the COCO weights
    model.load_weights(COCO_MODEL_PATH, by_name=True,
                       exclude=["mrcnn_class_logits", "mrcnn_bbox_fc", 
                                "mrcnn_bbox", "mrcnn_mask"])
elif init_with == "last":
    # Load the last model you trained and continue training
    model.load_weights(model.find_last()[1], by_name=True)
    #model_path = "./logs/bowl20180323T1602/mask_rcnn_bowl_0100.h5"
    #model.load_weights(model_path, by_name=True)
    
# Training dataset
dataset_train = BowlDataset()
dataset_train.load_bowl('stage1_train')
dataset_train.prepare()

# # Validation dataset
dataset_val = BowlDataset()
dataset_val.load_bowl('stage1_train')
dataset_val.prepare()

# Train the head branches
# Passing layers="heads" freezes all layers except the head
# layers. You can also pass a regular expression to select
# which layers to train by name pattern.
#model.train(dataset_train, dataset_val, 
            #learning_rate=bowl_config.LEARNING_RATE, 
            #epochs=25, 
            #layers='heads')

model.train(dataset_train, dataset_val, 
            learning_rate=bowl_config.LEARNING_RATE / 10,
            epochs=100, 
            layers="all")

```

I always get one of these two errors:

![error](https://user-images.githubusercontent.com/19358803/37865223-aea4d27e-2f79-11e8-8691-cfecd1f6e7b7.PNG)
![error2](https://user-images.githubusercontent.com/19358803/37865264-7ca83fc6-2f7a-11e8-8fbf-b39f82a6aee7.PNG)

**Comment by killthekitten** [comment]
Sorry for the late reply.

1. I see you've tried to load the weights directly. Did it help?
2. The first two warnings are ok, you can ignore them. 
3. As for the `TypeError`... hard to say. Which versions of python/keras/tensorflow are you on? My first guess would be that `model.find_last()` works differently in windows environment but I can't find anything Linux-specific in it.

**Comment by ChristophNeuner** [comment]
1. No unfortunately this did not help either.
3. Python 3.5.4, keras 2.1.3, tensorflow 1.5.0
I ran the code on linux.
